plugins/modules/apache2_mod_proxy.py

```diff
@@ -388,64 +383,47 @@ def main():
-        module.exit_json(
-            changed=False,
-            members=json_output_list
-        )
-    else:
-        changed = False
-        member_exists = False
-        member_status = {'disabled': False, 'drained': False, 'hot_standby': False, 'ignore_errors': False}
-        for mode in member_status:
-            for state in states:
-                if mode == state:
-                    member_status[mode] = True
-                elif mode == 'disabled' and state == 'absent':
-                    member_status[mode] = True
-
-        for member in mybalancer.members:
-            if str(member.host) == module.params['member_host']:
-                member_exists = True
-                if module.params['state'] is not None:
-                    member_status_before = member.status
-                    if not module.check_mode:
-                        member_status_after = member.status = member_status
-                    else:
-                        member_status_after = member_status
-                    if member_status_before != member_status_after:
-                        changed = True
-                json_output = member.as_dict()
-        if member_exists:
-            module.exit_json(
-                changed=changed,
-                member=json_output
-            )
+        self.mybalancer = Balancer(self.module, self.vars.balancer_vhost, self.vars.balancer_url_suffix, tls=self.vars.tls)
+
+    def __run__(self):
+        if self.vars.member_host is None:
+            self.vars.members = [member.as_dict() for member in self.mybalancer.members]
         else:
-            module.fail_json(
-                msg='{member_host} is not a member of the balancer {balancer_vhost}!'.format(
-                    member_host=module.params['member_host'],
-                    balancer_vhost=module.params['balancer_vhost'],
-                )
-            )
+            member_exists = False
+            member_status = {'disabled': False, 'drained': False, 'hot_standby': False, 'ignore_errors': False}
+            for mode in member_status:
+                for state in self.vars.state or []:
+                    if mode == state:
+                        member_status[mode] = True
+                    elif mode == 'disabled' and state == 'absent':
+                        member_status[mode] = True
+
+            for member in self.mybalancer.members:
+                if str(member.host) == self.vars.member_host:
+                    member_exists = True
+                    if self.vars.state is not None:
+                        member_status_before = member.status
+                        if not self.check_mode:
+                            member_status_after = member.status = member_status
+                        else:
+                            member_status_after = member_status
+                        self.changed = (member_status_before != member_status_after)
```

Comment:
```suggestion
                        self.changed |= (member_status_before != member_status_after)
```
Otherwise you're overwriting `changed`.

(Note that in CI, both members have the same hostname, so the loop always matches both of them.)

Reply:
Well, you nailed it :-) Thanks! (at least in local the test passes, let's see how it goes in the CI)